docs/core/extensions/httpclient-factory.md

```diff
@@ -235,6 +243,41 @@ There are several additional configuration options for controlling the `IHttpCli
 | <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.RedactLoggedHeaders%2A> | Sets the collection of HTTP header names for which values should be redacted before logging. |
 | <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.SetHandlerLifetime%2A> | Sets the length of time that a `HttpMessageHandler` instance can be reused. Each named client can have its own configured handler lifetime value. |
 
+## Using HttpClientFactory together with SocketsHttpHandler
+
+Since .NET Core 2.1, `SocketsHttpHandler` implementation of `HttpMessageHandler` was added, that allows configuring `PooledConnectionLifetime`. This setting is used to ensure that the handler will react to DNS changes, so using `SocketsHttpHandler` is considered to be an alternative to using `IHttpClientFactory`, see [Guidelines for using HTTP clients](../../fundamentals/networking/http/httpclient-guidelines.md).
+
+However, `SocketsHttpHandler` and `IHttpClientFactory` can be used together, to benefit from configurability on both low level (e.g. using `LocalCertificateSelectionCallback` for dynamic certificate selection) and high level (e.g. leveraging DI integration and several client configurations).
+
+To do that, you need to
+1. Specify `SocketsHttpHandler` as `PrimaryHandler` and set up it's `PooledConnectionLifetime` (e.g. to a value that was previously in `HandlerLifetime`)
+2. As `SocketsHttpHandler` will handle connection pooling and recycling, then `IHttpClientFactory`-level handler recycling is not needed anymore, so it could be disabled by setting `HandlerLifetime` to `Timeout.InfiniteTimeSpan`.
+
+```csharp
+services.AddHttpClient(name)
+    .ConfigurePrimaryHttpMessageHandler(() =>
+    {
+        return new SocketsHttpHandler()
+        {
+            PooledConnectionLifetime = TimeSpan.FromMinutes(handlerLifetimeInMinutes)
```

Comment:
Nowhere 😅 kind of an example of thing that a user could have around. The code was mostly from https://github.com/dotnet/runtime/discussions/76064. I didn't want to use a specific constant there, but if you think it will be better, I can change it to `TimeSpan.FromMinutes(2)` for example.

Reply:
Yes, I think that would make a lot more sense. Thanks